mlflow/pyfunc/__init__.py

```diff
@@ -1015,6 +1015,22 @@ def _check_udf_return_type(data_type):
     return False
 
 
+def _is_spark_connect_session(spark):
+    import pyspark
+    from packaging.version import Version
+
+    if Version(pyspark.__version__) < Version("3.4.0"):
+        return False
+
+    try:
+        from pyspark.sql.connect.session import SparkSession
```

Comment:
The class module might be changed in spark 4.0 if we make a new pyspark-client package.

We can use `"SPARK_CONNECT_MODE_ENABLED" in os.environ` to check whether it is spark connect mode.

@hyukjinKwon WDYT ?

Reply:
Yes.

Reply:
sounds good